autoload/fern/internal/drawer/hover_popup.vim

```diff
@@ -0,0 +1,46 @@
+function! fern#internal#drawer#hover_popup#init() abort
+  if g:fern#disable_drawer_hover_popup
+    return
+  endif
+
+  if !exists('*popup_create')
+    return
+  endif
+
+  augroup fern_internal_drawer_hover_popup_init
+    autocmd! * <buffer>
+    autocmd CursorMoved <buffer> call s:cursor_moved_event()
+  augroup END
+endfunction
+
+function! s:cursor_moved_event() abort
+  let l:helper = fern#helper#new()
+
+  if s:should_display_popup()
+    call s:show_popup(helper)
+  endif
+endfunction
+
+function! s:show_popup(helper)
+  let l:node = a:helper.sync.get_cursor_node()
+  if l:node is# v:null
+    return
+  endif
+
+  " calculate the screen column of node label
+  let l:line = getline('.')
+  let l:labelbegin = charidx(l:line, strridx(l:line, l:node.label))
+  let l:labelbegin = l:labelbegin < 0 ? 0 : l:labelbegin
+  let l:col_offset = wincol() - getcursorcharpos()[2] + l:labelbegin
+
+  call popup_create(l:node.label, #{
```

Comment:
It was introduced in [4c6d90458](https://github.com/vim/vim/commit/4c6d90458baae843463f930fdc3fe4a7a2191d27) (patch 8.1.1705: using ~{} for a literal dict is not nice, 2019-07-16), but it was only introduced in v0.6.1 of Neovim. I'm going to revert to using the normal dictionary syntax though, in case anyone happens to have a functioning `popup_create()` in Neovim.